types/jsonwebtoken/index.d.ts

```diff
@@ -51,49 +51,49 @@ export interface SignOptions {
      * - ES512:    ECDSA using P-521 curve and SHA-512 hash algorithm
      * - none:     No digital signature or MAC value included
      */
-    algorithm?: Algorithm;
-    keyid?: string;
+    algorithm?: Algorithm | undefined;
+    keyid?: string | undefined;
     /** expressed in seconds or a string describing a time span [zeit/ms](https://github.com/zeit/ms.js).  Eg: 60, "2 days", "10h", "7d" */
-    expiresIn?: string | number;
+    expiresIn?: string | number | undefined;
```

Comment:
@sandersn It's probably good to reconsider this change to restore type `string | number`, since the `expiresIn` cannot be undefined after release [8.0.0](https://github.com/auth0/node-jsonwebtoken/compare/v7.4.3...v8.0.0).
📓 According to [this PR](https://github.com/auth0/node-jsonwebtoken/pull/348), the `undefined` will throw validation error:
"expiresIn" should be a number of seconds or string representing a timespan'.